hexrd/ui/import_data_panel.py

```diff
@@ -0,0 +1,247 @@
+
+            files, manual = ImageLoadManager().load_images([selected_file])
+            dialog = LoadImagesDialog(files, manual, self.ui.parent())
+            if not dialog.exec_():
+                return
+
+            # If it is a hdf5 file allow the user to select the path
+            ext = os.path.splitext(selected_file)[1]
+            if (ImageFileManager().is_hdf5(ext) and not
+                    ImageFileManager().path_exists(selected_file)):
+                ImageFileManager().path_prompt(selected_file)
+
+            ImageLoadManager().read_data(files, parent=self.ui)
+
+            file_names = [os.path.split(f[0])[1] for f in files]
+            self.ui.files_label.setText(', '.join(file_names))
+            self.ui.outline.setEnabled(True)
+            self.ui.add_template.setEnabled(True)
+            self.ui.save.setDisabled(True)
+            self.ui.transforms.setEnabled(True)
+            self.ui.add_transform.setEnabled(True)
+            self.parent().action_show_toolbar.setChecked(False)
+            self.parent().action_show_toolbar.setDisabled(True)
+
+    def add_transform(self):
+        ilm = ImageLoadManager()
+        state = HexrdConfig().load_panel_state
+        ilm.set_state({ 'trans': [self.ui.transforms.currentIndex()] })
+        ilm.begin_processing(postprocess=True)
+        self.ui.transforms.setCurrentIndex(0)
+
+    def add_template(self):
+        det = self.ui.detectors.currentText()
+        self.it = InteractiveTemplate(
+            HexrdConfig().image('default', 0), self.parent())
+        self.it.create_shape(module=self.mod, file_name=det + '.txt')
+        self.ui.add_template.setDisabled(True)
+        self.ui.trans.setEnabled(True)
+        self.ui.trans.setChecked(True)
+        self.ui.rotate.setEnabled(True)
+        self.ui.button_box.setEnabled(True)
+        self.ui.detectors.setDisabled(True)
+        self.ui.load.setDisabled(True)
+        self.ui.complete.setEnabled(True)
+
+    def setup_translate(self):
+        if self.it is not None:
+            self.it.disconnect_rotate()
+            self.it.connect_translate()
+
+    def setup_rotate(self):
+        if self.it is not None:
+            self.it.disconnect_translate()
+            self.it.connect_rotate()
+
+    def clear_boundry(self):
+        if self.it is None:
+            return
+        self.it.clear()
+        self.it = None
+
+    def crop_and_mask(self):
+        if self.ui.trans.isChecked():
+            self.it.disconnect_translate()
+        else:
+            self.it.disconnect_rotate()
+        img = self.it.get_mask()
+        bounds = np.array([int(val) for val in self.it.crop()]).reshape(2, 2)
+        self.finalize(img, bounds)
+        self.completed_detectors.append(self.ui.detectors.currentText())
+        self.ui.trans.setDisabled(True)
+        self.ui.rotate.setDisabled(True)
+        self.ui.button_box.setDisabled(True)
+        self.ui.add_template.setDisabled(True)
```

Comment:
We might be able to make this more concise with loop?

```python
for w in [self.ui.trans, self.ui.rotate, self.ui.button_box, self.ui.add_template]:
    w.setDisabled(True)
```

Reply:
There are a couple of places where multiple widgets are being enabled/disabled at once (like [here](https://github.com/HEXRD/hexrdgui/blob/ebbc5e368788e2ba4b9a1480a4a1669044890624/hexrd/ui/import_data_panel.py#L189-L196), [here](https://github.com/HEXRD/hexrdgui/blob/ebbc5e368788e2ba4b9a1480a4a1669044890624/hexrd/ui/import_data_panel.py#L105-L111), and [here](https://github.com/HEXRD/hexrdgui/blob/ebbc5e368788e2ba4b9a1480a4a1669044890624/hexrd/ui/import_data_panel.py#L126-L132)). This loop could be factored out into a function and reused, unless that may make things more confusing? For example:
```
def set_enabled(self, widgets, enabled):
  for w in widgets:
    w.setEnabled(enabled)

self.setEnabled([self.ui.trans, self.ui.add_template,
                 self.ui.button_box, self.ui.rotate], False)
```

Reply:
Perfect, even better!